src/utils/user-utils.ts

```diff
@@ -1,12 +1,21 @@
 import type { User } from "src/generated/homeLambdasClient/models/User";
 import config from "src/app/config";
+import strings from "src/localization/strings";
 
 /**
- * To get the severa user id when logged in, if severaUserId is not found, it will return the test user severa id
+ * To get the severa user id when logged in, if severaUserId is not found and in develop mode, it will return the test user severa id
  *
  * @param user
  * @returns user severaUserId or testUserSeveraId
  */
 export const getSeveraUserId = (user: User | undefined): string => {
-  return user?.attributes?.severaUserId ?? config.user.testUserSeveraId ?? "";
+  const severaUserId = user?.attributes?.severaUserId;
+
+  if (!severaUserId) {
+    if (import.meta.env.MODE === "development") {
+      return config.user.testUserSeveraId ?? "";
+    }
+    throw new Error(strings.error.noSeveraUserId);
```

Comment:
Throwing the error like this seems to crash the whole app. I would suggest using the ErrorHandler like we do in other places, also consider in this case what should be rendered IN the balance card when there is no data and once the error handler has been closed. Perhaps a message "Something went wrong getting your balance"